src/main/java/bio/terra/grammar/azure/SynapseVisitor.java

```diff
@@ -4,37 +4,38 @@
 import bio.terra.grammar.SQLParser;
 import bio.terra.model.DatasetModel;
 import bio.terra.service.resourcemanagement.azure.AzureStorageAccountResource.FolderType;
+import bio.terra.service.snapshotbuilder.query.TableNameGenerator;
 import java.util.Map;
 import java.util.Objects;
 
 public class SynapseVisitor extends DatasetAwareVisitor {
   private final String sourceDatasetDatasource;
 
+  private static final String azureTableName =
+      """
+          (SELECT * FROM
+          OPENROWSET(
+            BULK '%s',
+            DATA_SOURCE = '%s',
+            FORMAT = 'parquet') AS %s)
+          """;
+
   public SynapseVisitor(Map<String, DatasetModel> datasetMap, String sourceDatasetDatasource) {
     super(datasetMap);
     this.sourceDatasetDatasource = sourceDatasetDatasource;
   }
 
-  public String generateAlias(String tableName) {
+  public static String generateAlias(String tableName) {
     return "alias" + Math.abs(Objects.hash(tableName));
   }
 
   @Override
   public String visitTable_expr(SQLParser.Table_exprContext ctx) {
     String tableName = getNameFromContext(ctx.table_name());
     String alias = generateAlias(tableName);
-    return """
-      (SELECT * FROM
-      OPENROWSET(
-        BULK '%s',
-        DATA_SOURCE = '%s',
-        FORMAT = 'parquet') AS %s) AS %s
-      """
-        .formatted(
-            FolderType.METADATA.getPath("parquet/%s/*/*.parquet".formatted(tableName)),
-            sourceDatasetDatasource,
-            "inner_" + alias,
-            alias);
+    String azureTableName = azureTableName(sourceDatasetDatasource).generate(tableName);
+    String azureTableNameAliased = azureTableName + " AS %s\n";
+    return azureTableNameAliased.formatted(alias);
```

Comment:
It seems a bit awkward to me to mix both string concatenation and formatting like this. Also, is there a reason not to call `generateTableName()` directly? How about
```suggestion
    return  "%s AS %s ".formatted(generateTableName(tableName, sourceDatasetDatasource), alias);
```

And is the `\n` needed here? I think all that's required is whitespace after the `alias`.